src/lib.rs

```diff
@@ -91,10 +91,14 @@ pub struct Assert {
     cmd: Vec<String>,
     expect_success: bool,
     expect_exit_code: Option<i32>,
-    expect_output: Option<String>,
-    fuzzy_output: bool,
-    expect_error_output: Option<String>,
-    fuzzy_error_output: bool,
+    expect_stdout: Option<OutputAssertion>,
+    expect_stderr: Option<OutputAssertion>,
+}
+
+#[derive(Debug)]
+struct OutputAssertion {
+    expect: String,
+    fuzzy: bool,
```

Comment:
I like the way you split out this struct, it's very neat.

Seeing how the code below with the matches on fuzziness etc. is pretty much duplicated, I'm wondering if maybe it can become a method on this type, like `OutputAssertion::compare_with(String)`? (It might also be a good idea to make this struct generic over a trait `OutputType `, but more on that in a comment below.)

At one point I was thinking about introducing an `enum Precision { Exact, Fuzzy }` instead of using a bool, but didn't go through with it.